### ISSUE TITLE
Could not resolve module @angular/core when using "files" in tsconfig.json

### ISSUE THREAD

**Issue by DutchKevv** [angular/angular#22427]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

When using "files" in my tsconfig.json to include a parent directory .ts file, it gives error "Could not resolve module @angular/core". When removing the "files" in tsconfig.json, all works fine again. I also tried including the src files etc, but nothing seems to work :( Thanks for any advice!

folder structure 
- shared
  - modules
    - coinpush
      - constant
        - index.ts
- client
  - package.json
  - tsconfig.json
  - src/etc
  - node_modules

```
{
  "compilerOptions": {
    "allowJs": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": false,
    "lib": [
      "es2017",
      "dom"
    ],
    "sourceMap": false,
    "target": "es2015",
    "module": "es2015",
    "moduleResolution": "node"
  },
  "files": [
    "../shared/modules/coinpush/constant/index.ts"
  ],
  "angularCompilerOptions": {
    "genDir": "./aot",
    "comments": false,
    "entryModule": "./src/app/app.module#AppModule",
    "preserveWhitespaces": false,
    "skipMetadataEmit": true
  }
}
```

```
ERROR in Error: Could not resolve module @angular/core
    at StaticSymbolResolver.getSymbolByModule (/Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler/bundles/compiler.umd.js:29766:30)
    at StaticReflector.findDeclaration (/Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler/bundles/compiler.umd.js:31635:63)
    at StaticReflector.initializeConversionMap (/Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler/bundles/compiler.umd.js:31953:36)
    at new StaticReflector (/Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler/bundles/compiler.umd.js:31571:14)
    at Object.createAotCompiler (/Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler/bundles/compiler.umd.js:32957:44)
    at AngularCompilerProgram._createCompiler (/Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler-cli/src/transformers/program.js:404:37)
    at AngularCompilerProgram.get [as hostAdapter] (/Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler-cli/src/transformers/program.js:327:22)
    at AngularCompilerProgram._createProgramWithBasicStubs (/Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler-cli/src/transformers/program.js:435:73)
    at /Users/kewinbrandsma/Projects/TradeJS/client/node_modules/@angular/compiler-cli/src/transformers/program.js:143:28
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

```

## Expected behavior
include parent directory .ts file and find node_modules/@angular/core

## Environment

<pre><code>
Angular version:5.2.6

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 8.9.4
- Platform:  Mac

p.s. I seem to have fixed it with the following tsconfig.json

```
{
  "compilerOptions": {
    "baseUrl": ".",
    "allowJs": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": false,
    "lib": [
      "es2017",
      "dom"
    ],
    "sourceMap": false,
    "target": "es2015",
    "module": "es2015",
    "moduleResolution": "node",
    "paths": {
      "@angular/*": [
        "../node_modules/@angular/*"
      ]
    }
  },
  "include": [
    "../../shared/modules/coinpush/constant/index.ts",
    "../**/*.ts"
  ],
  "angularCompilerOptions": {
    "genDir": "./aot",
    "comments": false,
    "entryModule": "./app/app.module#AppModule",
    "preserveWhitespaces": false,
    "skipMetadataEmit": true
  }
}
```

basically setting the paths + include, but it seems like a strange workaround for something I dont understand?

**Comment by alexeagle** [comment]
You don't say what command you ran that produces the error. Was it just `./node_modules/.bin/ngc` ?

**Comment by mhevery** [comment]
Closing do to inactivitiy

**Comment by tkw722** [comment]
I've seen this same issue, using either `"files": []` or `"include": []` in my tsconfig.json. Without either one of them, everything compiles fine, but when I have anything like that in there, I get: `ERROR in Could not resolve module @angular/core`. I get this when I run `ng serve`.

**Comment by adamgen** [comment]
> I've seen this same issue, using either `"files": []` or `"include": []` in my tsconfig.json. Without either one of them, everything compiles fine, but when I have anything like that in there, I get: `ERROR in Could not resolve module @angular/core`. I get this when I run `ng serve`.

I get the exact same behaviour

**Comment by orgertot** [comment]
> I've seen this same issue, using either `"files": []` or `"include": []` in my tsconfig.json. Without either one of them, everything compiles fine, but when I have anything like that in there, I get: `ERROR in Could not resolve module @angular/core`. I get this when I run `ng serve`.

Me too...

**Comment by CrazyBaran** [comment]
> I've seen this same issue, using either `"files": []` or `"include": []` in my tsconfig.json. Without either one of them, everything compiles fine, but when I have anything like that in there, I get: `ERROR in Could not resolve module @angular/core`. I get this when I run `ng serve`.

The same for me.
